codalab/worker_manager/worker_manager.py

```diff
@@ -21,6 +23,20 @@
 WorkerJob = namedtuple('WorkerJob', ['active'])
 
 
+def restart():
+    """
+    Restarts the current program, cleaning up file objects and descriptors
+    """
+    try:
+        p = psutil.Process(os.getpid())
+        for handler in p.open_files() + p.connections():
```

Comment:
What things are open that we need to cleanup?

Reply:
As of now, there shouldn't be any I think...but I added this because I figured it couldn't hurt / might future-proof things a bit. Also, even if this block completely fails, the restarting should still work fine.